### PR TITLE
[TACHYON-1592] Add unit test coverage for HdfsUnderFileSystem

### PR DESCRIPTION
**Pull request by luoli523**
There are really not that much utility methods in HdfsUnderFileSystem good for unit test. The apis of FileSystem are better tested by integration tests.

I added a little tests to check some configuration.

### CODE REVIEW

File: underfs/hdfs/src/test/java/tachyon/underfs/hdfs/HdfsUnderFileSystemTest.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+public final class HdfsUnderFileSystemTest {
+
+  private HdfsUnderFileSystem mMockHdfsUnderFileSystem;
+
+  @Before
+  public final void before() throws Exception {
+    mMockHdfsUnderFileSystem = new HdfsUnderFileSystem("file:///", new TachyonConf(), null);
+  }
+
+  @Test
```
Comment (apc999) [inline underfs/hdfs/src/test/java/tachyon/underfs/hdfs/HdfsUnderFileSystemTest.java]: could you put a brief description in javadoc for each test case ? you don't need to cover the exceptions as they are not very important for unit tests

**Comment by apc999** [comment]
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.2:compile (default-compile) on project tachyon-underfs-glusterfs: Compilation failure: Compilation failure:
[ERROR] /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/glusterfs/src/main/java/tachyon/underfs/glusterfs/GlusterFSUnderFileSystem.java:[72,12] method prepareConfiguration in class tachyon.underfs.hdfs.HdfsUnderFileSystem cannot be applied to given types;
[ERROR] required: tachyon.conf.TachyonConf,org.apache.hadoop.conf.Configuration
[ERROR] found: java.lang.String,tachyon.conf.TachyonConf,org.apache.hadoop.conf.Configuration
[ERROR] reason: actual and formal argument lists differ in length
[ERROR] /home/jenkins/workspace/Tachyon-Pull-Request-Builder/underfs/glusterfs/src/main/java/tachyon/underfs/glusterfs/GlusterFSUnderFileSystem.java:[56,3] method does not override or implement a method from a supertype

File: underfs/hdfs/src/main/java/tachyon/underfs/hdfs/HdfsUnderFileSystem.java [hunk 2]
```diff
@@ -100,7 +100,7 @@ public UnderFSType getUnderFSType() {
    * configuration necessary for obtaining a usable {@linkplain FileSystem} instance.
    * </p>
    *
-   * @param path file system path
+   * @param tachyonConf Tachyon Configuration
```
Comment (yupeng9) [inline underfs/hdfs/src/main/java/tachyon/underfs/hdfs/HdfsUnderFileSystem.java]: Could you also keep the param path?

File: underfs/hdfs/src/test/java/tachyon/underfs/hdfs/HdfsUnderFileSystemTest.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  public void getUnderFSTypeTest() throws Exception {
+    Assert.assertEquals(UnderFSType.HDFS, mMockHdfsUnderFileSystem.getUnderFSType());
+  }
+
+  /**
+   * Tests the
+   * {@link HdfsUnderFileSystem#prepareConfiguration(String, TachyonConf, Configuration)} method.
+   *
+   * Check the hdfs implements class and tachyon underfs config setting
```
Comment (yupeng9) [inline underfs/hdfs/src/test/java/tachyon/underfs/hdfs/HdfsUnderFileSystemTest.java]: Check -> Checks

**Comment by yupeng9** [comment]
One way to test the additional methods, is to mock the wrapped `mFs` (via Whitebox), and verify the right methods are called. But not sure too much value of this.

**Comment by luoli523** [comment]
@yupeng9  I tried the mock mFs using Whitebox but that's really no use to do that in the unit test. So I just test the utility methods.

**Comment by yupeng9** [comment]
LGTM. Thanks for adding the test coverage.

**Pull request merged**